app/assets/javascripts/admin/addon/components/admin-page-header.gjs

```diff
@@ -0,0 +1,58 @@
+import { hash } from "@ember/helper";
+import { htmlSafe } from "@ember/template";
+import DBreadcrumbsContainer from "discourse/components/d-breadcrumbs-container";
+import DBreadcrumbsItem from "discourse/components/d-breadcrumbs-item";
+import HorizontalOverflowNav from "discourse/components/horizontal-overflow-nav";
+import i18n from "discourse-common/helpers/i18n";
+import AdminPageActionButton from "admin/components/admin-page-action-button";
+
+const AdminPageHeader = <template>
+  <div class="admin-page-header">
+    <div class="admin-page-header__breadcrumbs">
+      <DBreadcrumbsContainer />
+      <DBreadcrumbsItem @path="/admin" @label={{i18n "admin_title"}} />
+      {{yield to="breadcrumbs"}}
+    </div>
+
+    <div class="admin-page-header__title-row">
+      {{#if @titleLabel}}
+        <h1 class="admin-page-header__title">{{i18n @titleLabel}}</h1>
+      {{/if}}
+      <div class="admin-page-header__actions">
+        {{yield
```

Comment:
@CvX before I go all in on this, can you please tell me if this is crazy or not? :sweat_smile: I based it on what FormKit is doing but maybe it's overkill in this case. Basically I want the API for app/assets/javascripts/admin/addon/components/admin-backups-actions.gjs and other admin headers, because I don't want people to have to remember to add the correct CSS classes for these header buttons. I want them to be consistent. But wrapping `DButton` with `AdminPageActionButton` also feels like overkill. What do you think, is there a nicer way to do this?

Reply:
I think it makes sense to provide context-specific button components. It could be simplified though
This could work:

```gjs
const ActionButton = <template>
  <DButton class="admin-page-action-button btn-small" ...attributes />
</template>

const PrimaryButton = <template><ActionButton class="btn-primary" ...attributes /></template>
const DangerButton = <template><ActionButton class="btn-danger" ...attributes /></template>
const DefaultButton = <template><ActionButton class="btn-default" ...attributes /></template>
```

and:

```hbs
{{yield
  (hash
    Primary=PrimaryButton
    Danger=DangerButton
    Default=DefaultButton
  )
  to="actions"
}}
```

and then e.g.

```hbs
<@actions.Default
  @label="admin.backups.read_only.enable.label"
  class="admin-backups__toggle-read-only"
  {{! instead of @additionalClasses }}
/>
```

Reply:
@CvX this doesn't work unfortunately, here is what I have in the code copying your examples:

![image](https://github.com/user-attachments/assets/341a98cf-804c-4009-a765-1e4df4a2b941)

And here is the UI:

![image](https://github.com/user-attachments/assets/bd80b1c4-f67b-458f-aa0d-2c3f368f01d9)

It's like the `@` arguments like `@label` aren't being passed down? No console errors or anything. If I change it to manually specify which `@` arguments are used it works correctly, this feels kinda ugly though (though it is what I had originally):

![image](https://github.com/user-attachments/assets/0ff07535-91ae-409c-8ca2-56c8b9171529)

Mostly I'm surprised Ember doesn't have a way to forward all `@arguments` down to the next component

Reply:
Ah, derp, of course 😅 Yes, splarguments are still not supported, you do have to pass them one by one 🥲 (see: https://github.com/emberjs/rfcs/pull/593)

Reply:
God can't wait for that, that will improve things so much.